### PR TITLE
prevent fk pointing to a time partition shard

### PR DESCRIPTION
**Pull request by dorinhogea**
Creating a time partition using a source table that has reverse foreign keys is an error.
Altering a table to point to a shard table is an error.

### CODE REVIEW

File: tests/timepart_constraints.test/runtest.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
```
Comment (adizaimi) [inline tests/timepart_constraints.test/runtest.sh]: This script does not need to be checked in.

File: tests/timepart_constraints.test/runit [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+if (( $? != 0 )) ; then
+   echo "FAILURE to create a normal partition"
+   exit 1
+fi
+
+timepart_stats
+
+# test 3: check attempt to drop a shard
+shard=`cdb2sql -tag ${CDB2_OPTIONS} $dbname default 'select shardname from comdb2_timepartshards limit 1'`
+cdb2sql ${CDB2_OPTIONS} $dbname default "drop table ${shard}" >> $OUT
```
Comment (adizaimi) [inline tests/timepart_constraints.test/runit]: Just wondering if we could capture the returned error message and check that it also says what we want it to say.

File: tests/timepart_constraints.test/runtest.sh (hunk 1 above)
Comment (dorinhogea) [reply tests/timepart_constraints.test/runtest.sh]: removed

**Comment by dorinhogea** [comment]
the whole bpfunc error mechanism needs to be re-addressed; we will do that in another PR.

**Pull request merged**